daemon/containerd/image_list.go

```diff
@@ -140,43 +141,31 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 }
 
 func (i *ImageService) singlePlatformImage(ctx context.Context, contentStore content.Store, image *ImageManifest) (*types.ImageSummary, []digest.Digest, error) {
-	diffIDs, err := image.RootFS(ctx)
+	contentSize, err := image.Size(ctx)
 	if err != nil {
 		return nil, nil, err
 	}
-	chainIDs := identity.ChainIDs(diffIDs)
 
-	size, err := image.Size(ctx)
+	diffIDs, err := image.RootFS(ctx)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, errors.Wrapf(err, "failed to get rootfs of image %s", image.Name())
 	}
 
 	// TODO(thaJeztah): do we need to take multiple snapshotters into account? See https://github.com/moby/moby/issues/45273
 	snapshotter := i.client.SnapshotService(i.snapshotter)
-	sizeCache := make(map[digest.Digest]int64)
 
-	snapshotSizeFn := func(d digest.Digest) (int64, error) {
-		if s, ok := sizeCache[d]; ok {
-			return s, nil
-		}
-		usage, err := snapshotter.Usage(ctx, d.String())
-		if err != nil {
-			if cerrdefs.IsNotFound(err) {
-				return 0, nil
-			}
-			return 0, err
-		}
-		sizeCache[d] = usage.Size
-		return usage.Size, nil
-	}
-	snapshotSize, err := computeSnapshotSize(chainIDs, snapshotSizeFn)
+	imageSnapshotID := identity.ChainID(diffIDs).String()
+	unpackedUsage, err := calculateSnapshotTotalUsage(ctx, snapshotter, imageSnapshotID)
 	if err != nil {
-		return nil, nil, err
+		if !cerrdefs.IsNotFound(err) {
+			return nil, nil, errors.Wrapf(err, "failed to calculate size of unpacked image %s", image.Name())
+		}
+		unpackedUsage = snapshots.Usage{Size: 0}
```

Comment:
Is the size critical for us to continue? Given my experience with the linked issue, I'm wondering if we'd still be able to "function" without this information, and (potentially) report this error as a warning to the user, but still continue.

Reply:
I think it's fine to continue without the size and we could probably just log a warning here. Although if the error here is not NotFound, then there's probably something really wrong and it will just pop up in a different place.

Reply:
Done.